lib/ruby_lsp/document.rb

```diff
@@ -80,7 +76,7 @@ def cache_get(request_name)
       @cache[request_name]
     end
 
-    sig { params(edits: T::Array[EditShape], version: Integer).void }
+    sig { params(edits: T::Array[T::Hash[Symbol, T.untyped]], version: Integer).void }
```

Comment:
I wonder if we could create a real `Edit` class instead?

Reply:
We receive these parameters from the `language-server_protocol` gem as hashes, so that would be a much more involved refactor. We'd need to start instantiating specialized objects for each type of request.

To be honest, I think it's a worth while contribution for the gem since passing hashes around is not very type friendly. But it'll take considerably more effort than this

Reply:
Yes, I've been meaning to do this refactor for the gem, actually. Maybe I can get this done this week.